src/resources/ble_esp32.cc

```diff
@@ -1504,7 +1607,7 @@ PRIMITIVE(close) {
 PRIMITIVE(scan_start) {
   ARGS(BleCentralManagerResource, central_manager, int64, duration_us)
 
-  Locker locker(BleResourceGroup::instance()->mutex());
+  Locker locker(ble_mutex);
```

Comment:
It smells a bit like it would be nice if we got the mutex from an object, because we need it to be non-null here. It's a bit too hard to see why it cannot be null.

Repeated in a few places.

Reply:
I will do that in a follow-up PR: move it to the resource-group.